### PR TITLE
List All Tabs / Visibility Bar Tests

### PR DESCRIPTION
**Pull request by ben-c-at-moz**
# Description

Add the Visibility Bar test as described in https://testrail.stage.mozaws.net/index.php?/cases/view/134655

## Bugzilla bug ID

**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1895905

## Type of change

Please delete options that are not relevant.

- [x] New Test
- [x] New POM

# How does this resolve / make progress on that bug?

Creates additional POM methods and selectors for the Tab Bar, creates a test that executes the steps in the TR test plan.

# Screenshot

Screenshot 2024-06-04 at 16 34 24

# Comments / Concerns

This is highly screen size dependent, may be difficult to calibrate across resolutions.

### CODE REVIEW

File: tests/tabs/test_list_all_tabs.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from selenium.webdriver import Firefox
+from selenium.webdriver.common.keys import Keys
+
+from modules.browser_object import TabBar
+
+
+def test_list_all_tabs(driver: Firefox, screenshot):
+    # C134655
```
Comment (sarinali) [inline tests/tabs/test_list_all_tabs.py:8]: i think we decided to do """""" type docs on test. also should indicate briefly what it does like other files

**Review by sarinali — CHANGES_REQUESTED** [review]
nice hack lol rmb to follow the style guidelines that you set tho 😭

File: modules/browser_object_tabbar.py [hunk 2]
```diff
@@ -88,3 +104,76 @@ def expect_tab_sound_status(
...
+    def scroll_tabs(self, direction: ScrollDirection) -> BasePage:
+        """Scroll tabs in tab bar using the < and > scroll buttons"""
+        logging.info(f"Scrolling tabs {direction}")
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            scroll_button = self.get_element(f"tab-scrollbox-{direction}-button")
+            scroll_button.click()
+        return self
+
+    def get_text_of_all_tabs_entry(self, selected=False, index=0) -> str:
+        """
```
[the remark this answers is not in the document]
Comment (ben-c-at-moz) [reply modules/browser_object_tabbar.py:137]: ehhhhhh, I did say that for small functions you don't have to go all out, I think I'll leave this one

File: tests/tabs/test_list_all_tabs.py (hunk 1 above)
Comment (ben-c-at-moz) [reply tests/tabs/test_list_all_tabs.py:8]: On it

File: modules/browser_object_tabbar.py [hunk 3]
```diff
@@ -88,3 +104,76 @@ def expect_tab_sound_status(
...
+        else:
+            entries = self.get_elements("all-tabs-entry")
+            entry = entries[index]
+        return entry.find_element(By.CLASS_NAME, "all-tabs-button").location
+
+    def scroll_on_all_tabs_menu(self, down=True, pixels=200) -> BasePage:
+        """Scroll the all tabs menu down or up."""
+        with self.driver.context(self.driver.CONTEXT_CHROME):
+            menu = self.get_element("all-tabs-menu")
+            # HACK: Can't figure out what the scrollbox selector is, but it's ~4 pixels
```
[the remark this answers is not in the document]
Comment (ben-c-at-moz) [reply modules/browser_object_tabbar.py:168]: YOLO

**Review by sarinali — APPROVED** [review]

**Pull request merged**